core/domain/activity_jobs_one_off.py

```diff
@@ -42,31 +42,44 @@ def entity_classes_to_map_over(cls):
     def map(model):
         ids_set = set(model.contributor_ids)
         summary_set = set(model.contributors_summary)
+        modified = False
         if len(ids_set) != len(model.contributor_ids):
             # When the contributor_ids contain duplicate ids.
+            modified = True
+            model.contributor_ids = list(ids_set)
             yield (
                 'DUPLICATE_IDS',
                 (model.id, model.contributor_ids, model.contributors_summary)
             )
         if ids_set - summary_set:
             # When the contributor_ids contain id that is not in
             # contributors_summary.
+            modified = True
```

Comment:
Instead of all this, wouldn't it be better to just use exp_services.compute_summary_of_exploration? If the summary is corrupt here then there might be issues there also.

Plus we should audit that code to make sure it's actually producing the correct thing.

Reply:
Done.